### ISSUE TITLE
Item sheet fails to parse, even when generated with updated definitions

### ISSUE THREAD

**Issue by ff-meli** [NotAdam/Lumina#2]
```
System.InvalidCastException
  HResult=0x80004002
  Message=Unable to cast object of type 'System.Int16' to type 'System.Byte'.
  Source=Lumina
  StackTrace:
   at Lumina.Excel.RowParser.ReadField[T](ExcelColumnDataType type) in Lumina\Lumina\Excel\RowParser.cs:line 215
   at Lumina.Excel.RowParser.ReadOffset[T](Int32 offset, Byte bit) in Lumina\Lumina\Excel\RowParser.cs:line 245
   at Lumina.Excel.GeneratedSheets.Item.PopulateData(RowParser parser, Lumina lumina) in Lumina\Lumina\Excel\GeneratedSheets\I\Item.cs:line 356
   at Lumina.Excel.ExcelSheet`1.GetRows() in Lumina\Lumina\Excel\ExcelSheet.cs:line 160
```

This has happened for a while with the latest lumina version.

I tried updating definitions from a more recent godbert build and manually exporting a new Item sheet via the spaghetti generator, with the same result.

I also tried manually hacking that particular set of bytes for unknown58, but it then failed on the next similar block (unknown60).

**Comment by NotAdam** [comment]
Yeah I think there's a bunch wrong with how I handle packed bools. The plan for this was to just replace it with column reads for generated files and also move it to a different nuget package so the lib itself isn't massive because of all the generated garbage. Just haven't got around to doing this yet

**Comment by NotAdam** [comment]
This was 'fixed' in 1.1.3. Casts will now silently fail and the value will be set to the default value unless you change `Options.ExcelSheetStrictCastingEnabled` to `true`, which will then revert to the original behaviour and throw exceptions. Ideally I'd like to expose more of the internal workings to the lib consumer, potentially just via logging so at least you can see something is happening rather than nothing.

I think this should cover most cases without causing too much trouble but let me know if we should do something different in this case
